uber/site_sections/registration.py

```diff
@@ -804,12 +804,18 @@ def review(self, session):
     def discount(self, session, message='', **params):
         attendee = session.attendee(params)
         if 'first_name' in params:
+            if 'overridden_price' in params:
+                if isinstance(attendee.overridden_price, str):
```

Comment:
HTML forms always pass a string - did you test this to make sure that `isinstance` is actually returning true as long as overridden_price can be typecast to an integer? The Python docs don't really indicate that the function would do that, but I'm not familiar enough to be sure (and it may be being typecast to `int` elsewhere, like when you load the session).

Alternatively, you can try to typecast to an integer and use a try/except block instead. So like:
```
try:
attendee.overridden_price = int(attendee.overridden_price)
except ValueError:
attendee.overridden_price = -1
```